presto-main/src/main/java/com/facebook/presto/operator/SimplePagesHashStrategy.java

```diff
@@ -200,6 +200,16 @@ public boolean positionEqualsPosition(int leftBlockIndex, int leftPosition, int
     @Override
     public boolean applyFilterFunction(int leftBlockIndex, int leftPosition, int rightPosition, Block[] allRightBlocks)
     {
-        return filterFunction.get().filter(leftPosition, channelArrays.get(leftBlockIndex), rightPosition, allRightBlocks);
+        return filterFunction.get().filter(leftPosition, getLeftBlocks(leftBlockIndex), rightPosition, allRightBlocks);
+    }
+
+    private Block[] getLeftBlocks(int leftBlockIndex)
+    {
+        if (channelArrays.isEmpty()) {
+            return EMPTY_BLOCK_ARRAY;
```

Comment:
I think the main concern here is that the PagesIndex class (which generates the block and position address) is not precisely designed to be used in this way with no columns (@dain can weigh in here with more details) and the leftBlockIndex and leftPosition being used is somewhat bogus. However, it just happens to work in this case.

For the sake of this PR, I think this should be ok for now. But we should work on the longer term fix with the new NestedLoopJoin as soon as possible.